pkg/cmd/join/exec.go

```diff
@@ -90,18 +92,22 @@ func (o *Options) run() error {
 		"join/service_account.yaml",
 	}
 
-	err = helpers.ApplyDirectly(clientHolder, reader, scenarioDirectory, o.values, files...)
+	err = apply.ApplyDirectly(clientHolder, reader, o.values, "", files...)
 	if err != nil {
 		return err
 	}
 
-	err = helpers.ApplyDeployment(kubeClient, reader, scenarioDirectory, o.values, "join/operator.yaml")
+	err = apply.ApplyDeployment(kubeClient, reader, o.values, "", "join/operator.yaml")
 	if err != nil {
 		return err
 	}
 
+	//quick fix for https://github.com/open-cluster-management-io/clusteradm/issues/12
+	fmt.Printf("Wait 10 sec... for the crd to be effective\n")
```

Comment:
you could use wait.Until here to get crd  of clustermanager.

Reply:
implement retry.OnError